### ISSUE TITLE
aria-rowcount and aria-colcount not being announced for element with role="grid"

### ISSUE THREAD

**Issue by gportela85** [nvaccess/nvda#14332]
### Steps to reproduce:
1 - Open https://plnkr.co/edit/2P724oRsPsm3X2Jl?open=main.js

Click on the first cell (United States on the top left)

### Actual behavior:
NVDA does not announce row-count and col-count

table
United States  row 2  Country  column 1

### Expected behavior:
NVDA to row-count and col-count
table  with 8619 rows and 7 columns
United States  row 2  Country  column 1

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge 107.0.1418.26 (Official build) (64-bit)

**Comment by feerrenrut** [comment]
- Because the data is lazy loaded in this sample, we'll need more precise instructions for interacting with the table. The timing will affect how it is read.
- When testing this ourselves, we initially had `table 1 row` reported, then navigating out of the table and then returning NVDA reported `table 8619 rows`
- We need to know how you navigated to the table, this information would be included in an NVDA debug log along with other details that will help us to understand the problem. Please reproduce this issue and provide a [log file](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps) of the behavior. Ensure your log level is set to debug in general preferences.

**Comment by gportela85** [comment]
@feerrenrut I understand that the data is being lazy loaded, but the values of `aria-colcount` and `aria-rowcount` are being manually set, so I would expect these value to be announced and the lazy loaded rows to be ignored. Is that not correct?

I tried navigating into the table three different ways:

First - Click on code code editor on the left, and then mouse click on a table cell.

Second - Click on the white background on the right, then click on cell.

(The two different ways above were just to check if there was a difference between focus initially coming from a different iFrame).

Third - Click on the back background on the right, and then press TAB to focus the grid header.

**Comment by Adriani90** [comment]
Ok I managed to reproduce this in every browser very simple in focus mode, but this issue is not reproducible in Browse mode.
1. Go to this link
https://plnkr.co/edit/2P724oRsPsm3X2Jl?open=main.js&preview
2. Switch to focus mode with nvda+space bar
3. Press tab until you land on the table or click with the mouse into a cell

Actual: NVDA reports an URL, the something with Javascript, and then the word "table", followed by the current content of focused cell and the cell coordinates.

Expected: NVDA should report table with y rows and x columns, focused cell content, focused cell coordinate.
